pkg/build/api/validation/validation.go

```diff
@@ -204,6 +205,20 @@ func validateSource(input *buildapi.BuildSource) fielderrors.ValidationErrorList
 	return allErrs
 }
 
+// validateBuildSourceType checks if the source type for this
+// build is one we understand.
+func validateBuildSourceType(sourceType buildapi.BuildSourceType) fielderrors.ValidationErrorList {
+	allErrs := fielderrors.ValidationErrorList{}
+	switch sourceType {
+	case buildapi.BuildSourceGit:
+	case buildapi.BuildSourceDockerfile:
+	case buildapi.BuildSourceBinary:
+	default:
+		allErrs = append(allErrs, fielderrors.NewFieldInvalid("type", sourceType, fmt.Sprintf("source type must be one of Git, Dockerfile, or Binary")))
```

Comment:
Gives the wrong error for build source type - should be required, not invalid, when empty string.